code/game/objects/items/trader.dm

```diff
@@ -1264,6 +1263,221 @@ var/global/list/alcatraz_stuff = list(
+	icon = 'icons/obj/device.dmi'
+	icon_state = "airprojector"
+	var/list/projected = list()
+	var/max_proj = 6
+
+/obj/item/airshield_projector/preattack(atom/target, mob/user , proximity)
+	var/turf/to_shield = get_turf(target)
+	if(projected.len < max_proj && istype(to_shield) && (!locate(/obj/effect/airshield) in to_shield))
+		playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+		var/obj/effect/airshield/A = new(to_shield)
+		A.owner = src
+		projected += A
+		visible_message("<span class='notice'>\The [user] deploys \the [A].</span>")
+		return TRUE
+	return FALSE
+
+//not to be confused with the structure in airshield.dm
+/obj/effect/airshield
+	name = "airshield"
+	desc = "A shield that allows only non-gasses to pass through."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "planner"
+	opacity = FALSE
+	mouse_opacity = FALSE
+	density = FALSE
+	anchored = TRUE
+	plane = ABOVE_HUMAN_PLANE
+	maptext_x = 11
+	maptext_y = 8
+	var/obj/item/airshield_projector/owner
+	var/life = 9
+
+/obj/effect/airshield/New()
+	..()
+	countdown()
+
+/obj/effect/airshield/proc/countdown()
+	maptext = "<span style=\"color:#FF8C00;font-size:12px;\">[life]</span>"
+	spawn(1 SECONDS)
+		life--
+		if(life>0)
+			countdown()
+		else
+			if(owner)
+				owner.projected -= src
+			update_nearby_tiles(loc)
+			qdel(src)
+
+/obj/effect/airshield/Cross(atom/movable/mover, turf/target, height=1.5, air_group = 0)
+	if(istype(mover))
+		return ..()
+	return FALSE
+
+var/list/decelerators = list()
+/obj/item/weapon/am_containment/decelerator
+	name = "antimatter decelerator"
+	desc = "Acts as a 'filter' to trap antiparticles emitted by radiation. In function, it can be used to power an antimatter engine and refuel itself with nearby radiation."
+
+/obj/item/weapon/am_containment/decelerator/New()
+	..()
+	decelerators += src
+
+/obj/item/weapon/am_containment/decelerator/Destroy()
+	decelerators -= src
+	..()
+
+/obj/item/weapon/am_containment/decelerator/proc/receive_pulse(power)
+	fuel = min(fuel_max, fuel + round(power/100))
+
+#define OMNIMODE_WIRE 0
+#define OMNIMODE_TOOL 1
+/obj/item/device/multitool/omnitool
+	name = "omnitool"
+	desc = "Combining the power of wirecutters and a multitool. For power cables, works as a multitool when you stand on top and use it. It also allows the user to remotely access APCs and air alarms."
+	icon_state = "omnitool"
+	origin_tech = Tc_ENGINEERING + "=4"
+	sharpness = 1
+	force = 6
+	var/mode = OMNIMODE_TOOL
+
+/obj/item/device/multitool/omnitool/attack_self(mob/user)
+	mode = !mode
+	to_chat(user, "<span class='notice'>You toggle the tool into [mode ? "multitool" : "wirecutter"] mode.</span>")
+
+/obj/item/device/multitool/omnitool/is_wirecutter()
+	return !mode
+
+/obj/item/device/multitool/omnitool/is_multitool()
+	return mode
+
+var/list/omnitoolable = list(/obj/machinery/alarm,/obj/machinery/power/apc)
```

Comment:
This could be a static var in the proc below, but if you think it's useful to have it available outside you can keep it this way.